File: 222/ws/src/demo_py_pkg/demo_py_pkg/person_node.py
import rclpy
from rclpy.node import Node
class PersonNode(Node):
    def __init__(self, node_name: str, name_value: str, age_value: int) -> None:
        print('PersonNode__init__被调用了')
        super().__init__(node_name)
        self.name = name_value
        self.age = age_value
    def eat(self,food_name: str):
        """
        方法：吃东西
        :food_name 参数：食物名称
        """
        #print(f'{self.name} ,{self.age}岁，爱吃{food_name}')
        self.get_logger().info(f'{self.name} ,{self.age}岁，爱吃{food_name}')

def main():
    rclpy.init()
    node = PersonNode('node_Vince','Vince', 22)
    node.eat('芒果')
    rclpy.spin(node)
    rclpy.shutdown()

File: 222/ws/src/demo_cpp_pkg/src/person_node.cpp
#include "rclcpp/rclcpp.hpp"

class PersonNode : public rclcpp::Node
{
    private:
        std::string name_;
        int age_;
    public:
        PersonNode(const std::string & node_name,const std::string & name_value, int age_value)
        :Node(node_name) /*调用父类构造函数，等于python中的super().__init__()*/ 
            {
                this->name_ = name_value;
                this->age_ = age_value;
            };
    void eat(const std::string & food_name)
    {
        RCLCPP_INFO(this->get_logger(),"我是%s,%d岁,爱吃%s",this->name_.c_str(),this->age_,food_name.c_str());
    };
};

int main(int argc,char** argv)
{
    rclcpp::init(argc,argv);
    auto node = std::make_shared<PersonNode>("person_node","Vince",22);
    node->eat("面包");
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}